File: examples/java/patterns/state/batch-result-pointers.java
BatchResult<String> results = context.map(
    "process-items",
    items,
    String.class,
    (ctx, item) -> {
        Object output = processItem(item);
        return storeOutput(output); // returns an S3 key, not the output itself
    });

// `results` carries pointers, not payloads.

File: examples/python/patterns/state/batch-result-pointers.py
@durable_step
def process(ctx: StepContext, item: dict) -> str:
    output = process_item(item)
    return store_output(output)  # returns an S3 key, not the output itself


results = context.map(items, process)
# `results` carries pointers, not payloads.
